server/services/expectedArrivalsService.ts

```diff
@@ -79,39 +71,68 @@ export default class ExpectedArrivalsService {
     const token = await this.hmppsAuthClient.getSystemClientToken(username)
     const welcomeClient = this.welcomeClientFactory(token)
 
+    return arrival.expected
+      ? this.confirmExpectedArrival(welcomeClient, prisonId, id, arrival)
+      : this.confirmUnexpectedArrival(welcomeClient, prisonId, arrival)
+  }
```

Comment:
little bit of refactoring here as we were trying to look up an unexpected arrival when by definition we don't have an arrival record for them.